public/app/plugins/datasource/loki/tracking.ts

```diff
@@ -0,0 +1,23 @@
+import { DataQueryResponse } from '@grafana/data';
+import { reportInteraction } from '@grafana/runtime';
+
+import { isLogsQuery, parseToArray } from './queryUtils';
+import { LokiQuery } from './types';
+
+export function trackQuery(response: DataQueryResponse, queries: LokiQuery[], app: string): void {
+  for (const query of queries) {
+    reportInteraction('grafana_loki_query_executed', {
+      app,
+      editor_mode: query.editorMode,
+      has_data: response.data.some((frame) => frame.length > 0),
+      has_error: response.error !== undefined,
+      legend: query.legendFormat,
+      line_limit: query.maxLines,
+      parsed_query: parseToArray(query.expr),
```

Comment:
Just thinking about how to save this. Wouldn't "just" string of these be easier to work with - to get most popular patterns. Not sure if you can group by on content of array?